finetuner-workflow/finetuner/inference.py

```diff
@@ -1,71 +1,77 @@
 import os
-import kserve
-import logging
+import uvicorn
 import torch
-from typing import Dict
-from transformers import pipeline, AutoModelForCausalLM, AutoTokenizer
 
-logging.basicConfig(level=kserve.constants.KSERVE_LOGLEVEL)
-logger = logging.getLogger(__name__)
+from pydantic import BaseModel
+from typing import Optional, List
+from fastapi import FastAPI
+from fastapi.middleware.cors import CORSMiddleware
+from transformers import pipeline
 
-options = {
-    'MODEL_PATH': os.getenv('MODEL_PATH'),
-    'MODEL_NAME': os.getenv('MODEL_NAME'),
-}
+class Completion(BaseModel):
+    prompt: str
+    engine: Optional[str] = None
+    max_new_tokens: Optional[int] = 10
+    temperature: Optional[float] = None
+    top_p: Optional[float] = None
+    top_k: Optional[int] = None
+    typical_p: Optional[float] = None
+    repetition_penalty: Optional[float] = None
+    do_sample: Optional[bool] = True
+    penalty_alpha: Optional[float] = None
+    num_return_sequences: Optional[int] = 1
+    stop_sequence: Optional[str] = None
+    bad_words: Optional[List] = None
+
+app = FastAPI(
+    title="Inference API"
+)
+
+app.add_middleware(
+    CORSMiddleware,
+    allow_origins=["*"],
+    allow_methods=["*"],
+    allow_headers=["*"],
+)
 
-default_params = {
-    'MIN_LENGTH': int(os.getenv('MIN_LENGTH', 1)),
-    'MAX_LENGTH': int(os.getenv('MAX_LENGTH', 100)),
-    'TEMPERATURE': float(os.getenv('TEMPERATURE', 1.0)),
-    'TOP_K': int(os.getenv('TOP_K', 50)),
-    'TOP_P': float(os.getenv('TOP_P', 0.95)),
-    'REPETITION_PENALTY': float(os.getenv('REPETITION_PENALTY', 1.0)),
-    'NUM_RETURN_SEQUENCES': int(os.getenv('NUM_RETURN_SEQUENCES', 1)),
+args = {
+    "model": os.getenv("INFERENCE_MODEL", "distilgpt2"),
+    "device": int(os.getenv("INFERENCE_DEVICE", 0)),
+    "port": int(os.getenv("INFERENCE_PORT", 80))
 }
```

Comment:
IMO using environment variables for this one while all of the other containers use command line arguments seems a little odd. Is there a particular reason to keep them this way, or just convenience?